src/Arcus.WebApi.Security/Authorization/JwtTokenAuthorizationFilter .cs

```diff
@@ -34,18 +45,80 @@ public JwtTokenAuthorizationFilter(JwtTokenAuthorizationOptions authorizationOpt
+            Guard.NotNull(context, nameof(context));
+            Guard.NotNull(context.HttpContext, nameof(context.HttpContext));
+            Guard.For<ArgumentException>(() => context.HttpContext.Request is null, "Invalid action context given without any HTTP request");
+            Guard.For<ArgumentException>(() => context.HttpContext.Request.Headers is null, "Invalid action context given without any HTTP request headers");
+            Guard.For<ArgumentException>(() => context.HttpContext.RequestServices is null, "Invalid action context given without any HTTP request services");
+
+            ILogger logger = context.HttpContext.RequestServices.GetLoggerOrDefault<JwtTokenAuthorizationFilter>();
+            IJwtTokenReader reader = _authorizationOptions.GetOrCreateJwtTokenReader(context.HttpContext.RequestServices);
+            
+            if (reader is null)
+            {
+                logger.LogError("Cannot validate JWT MSI token because no '{Type}' was registered in the options of the JWT authorization filter", nameof(IJwtTokenReader));
+                throw new InvalidOperationException("Cannot validate JWT MSI token because the registered JWT options were invalid");
+            }
+
             if (context.HttpContext.Request.Headers.TryGetValue(_authorizationOptions.HeaderName, out StringValues jwtString))
             {
-                bool isValidToken = await _authorizationOptions.JwtTokenReader.IsValidTokenAsync(jwtString);
-                if (string.IsNullOrWhiteSpace(jwtString) || isValidToken == false)
-                {
-                    context.Result = new UnauthorizedObjectResult("Unauthorized because of wrong JWT MSI token.");
-                }
+                await ValidateJwtTokenAsync(reader, context, jwtString, logger);
             }
             else
             {
-                context.Result = new UnauthorizedObjectResult("Unauthorized because of missing JWT MSI token header.");
+                LogSecurityEvent(logger, LogLevel.Error, "No JWT MSI token was specified in the request", HttpStatusCode.Unauthorized);
+                context.Result = new UnauthorizedObjectResult("No JWT MSI token header found in request");
+            }
+        }
+
+        private static async Task ValidateJwtTokenAsync(IJwtTokenReader reader, AuthorizationFilterContext context, StringValues jwtString, ILogger logger)
+        {
+            if (String.IsNullOrWhiteSpace(jwtString))
+            {
+                LogSecurityEvent(logger, LogLevel.Error, "Cannot validate JWT MSI token because the token is blank", HttpStatusCode.Unauthorized);
+                context.Result = new UnauthorizedObjectResult("Blank JWT MSI token");
+                
+                return;
             }
+
+            if (!JwtRegex.IsMatch(jwtString))
+            {
+                LogSecurityEvent(logger, LogLevel.Error, "Cannot validate JWT MSI token because the token is in an invalid format", HttpStatusCode.Unauthorized);
```

Comment:
```suggestion
                LogSecurityEvent(logger, LogLevel.Trace, "Cannot validate JWT MSI token because the token is in an invalid format", HttpStatusCode.Unauthorized);
```